code/game/machinery/vendors/generic_vendors.dm

```diff
@@ -0,0 +1,867 @@
+	ads_list = list("Only the finest!","Have some tools.","The most robust equipment.","The finest gear in space!")
+	refill_canister = /obj/item/vending_refill/assist
+
+/obj/machinery/economy/vending/boozeomat
+	name = "\improper Booze-O-Mat"
+	desc = "A technological marvel, supposedly able to mix just the mixture you'd like to drink the moment you ask for one."
+	icon_state = "boozeomat"        //////////////18 drink entities below, plus the glasses, in case someone wants to edit the number of bottles
+	icon_deny = "boozeomat_deny"
+	icon_lightmask = "smartfridge"
+	icon_panel = "smartfridge"
+	icon_broken = "smartfridge"
+	products = list(/obj/item/reagent_containers/food/drinks/bottle/gin = 5,
+					/obj/item/reagent_containers/food/drinks/bottle/whiskey = 5,
+					/obj/item/reagent_containers/food/drinks/bottle/tequila = 5,
+					/obj/item/reagent_containers/food/drinks/bottle/vodka = 5,
+					/obj/item/reagent_containers/food/drinks/bottle/vermouth = 5,
+					/obj/item/reagent_containers/food/drinks/bottle/rum = 5,
+					/obj/item/reagent_containers/food/drinks/bottle/wine = 5,
+					/obj/item/reagent_containers/food/drinks/bag/goonbag = 3,
+					/obj/item/reagent_containers/food/drinks/bottle/cognac = 5,
+					/obj/item/reagent_containers/food/drinks/bottle/kahlua = 5,
+					/obj/item/reagent_containers/food/drinks/cans/beer = 6,
+					/obj/item/reagent_containers/food/drinks/cans/ale = 6,
+					/obj/item/reagent_containers/food/drinks/cans/synthanol = 15,
+					/obj/item/reagent_containers/food/drinks/bottle/orangejuice = 4,
+					/obj/item/reagent_containers/food/drinks/bottle/tomatojuice = 4,
+					/obj/item/reagent_containers/food/drinks/bottle/limejuice = 4,
+					/obj/item/reagent_containers/food/drinks/bottle/cream = 4,
+					/obj/item/reagent_containers/food/drinks/cans/tonic = 8,
+					/obj/item/reagent_containers/food/drinks/cans/cola = 8,
+					/obj/item/reagent_containers/food/drinks/cans/sodawater = 15,
+					/obj/item/reagent_containers/food/drinks/drinkingglass = 30,
+					/obj/item/reagent_containers/food/drinks/drinkingglass/shotglass = 30,
+					/obj/item/reagent_containers/food/drinks/ice = 9)
+	contraband = list(/obj/item/reagent_containers/food/drinks/tea = 10,
+					  /obj/item/reagent_containers/food/drinks/bottle/fernet = 5)
+	vend_delay = 15
+	slogan_list = list("I hope nobody asks me for a bloody cup o' tea...","Alcohol is humanity's friend. Would you abandon a friend?","Quite delighted to serve you!","Is nobody thirsty on this station?")
+	ads_list = list("Drink up!","Booze is good for you!","Alcohol is humanity's best friend.","Quite delighted to serve you!","Care for a nice, cold beer?","Nothing cures you like booze!","Have a sip!","Have a drink!","Have a beer!","Beer is good for you!","Only the finest alcohol!","Best quality booze since 2053!","Award-winning wine!","Maximum alcohol!","Man loves beer.","A toast for progress!")
```

Comment:
```suggestion
	slogan_list = list("I hope nobody asks me for a bloody cup o' tea...", "Alcohol is humanity's friend. Would you abandon a friend?", "Quite delighted to serve you!", "Is nobody thirsty on this station?")
	ads_list = list("Drink up!", "Booze is good for you!", "Alcohol is humanity's best friend.", "Quite delighted to serve you!", "Care for a nice, cold beer?", "Nothing cures you like booze!", "Have a sip!", "Have a drink!", "Have a beer!", "Beer is good for you!", "Only the finest alcohol!", "Best quality booze since 2053!", "Award-winning wine!", "Maximum alcohol!", "Man loves beer.", "A toast for progress!")
```
Prob are more like this